src/main/java/uk/co/ramp/EvaluateCase.java

```diff
@@ -0,0 +1,230 @@
+        switch (newStatus) {
+            case EXPOSED:
+                mean = properties.meanTimeToInfectious();
+                max = properties.maxTimeToInfectious();
+                break;
+            case EXPOSED_2:
+                mean = properties.meanTimeToInfected();
+                max = properties.maxTimeToInfected();
+                break;
+            default:
+                mean = properties.meanTimeToFinalState();
+                max = properties.maxTimeToFinalState();
+        }
+
+        if (newStatus != DEAD && newStatus != RECOVERED) {
+            timeToNextChange = getDistributionValue(mean, max);
+            p.setNextVirusStatusChange(currentTime + timeToNextChange);
+        } else {
+            p.setNextVirusStatusChange(Case.getDefault());
+        }
+
+        p.setStatus(newStatus);
+
+    }
+
+    public Set<Integer> checkActionsAtTimestep(int time) {
+        Set<Integer> alerts = new HashSet<>();
+        if (p.nextVirusStatusChange() == time) checkUpdateVirusStatus(time);
+        if (p.nextAlertStatusChange() == time) alerts = checkUpdateAlertStatus(time);
+
+        return alerts;
+    }
+
+    private void checkUpdateVirusStatus(int time) {
+
+        p.setNextVirusStatusChange(-1);
+        switch (p.status()) {
+            case EXPOSED:
+                updateVirusStatus(EXPOSED_2, time);
+                break;
+            case EXPOSED_2:
+                VirusStatus status = determineInfection(p);
+                updateVirusStatus(status, time);
+                if (status == INFECTED_SYMP) updateAlertStatus(REQUESTED_TEST, time);
+                break;
+            case INFECTED:
+                updateVirusStatus(RECOVERED, time);
+                break;
+            case INFECTED_SYMP:
+                status = determineOutcome(p);
+                updateVirusStatus(status, time);
+                break;
+            case SUSCEPTIBLE:
+            case RECOVERED:
+            case DEAD:
+                break;
+        }
+    }
+
+
+    private Set<Integer> checkUpdateAlertStatus(int time) {
+        LOGGER.trace("Changing alert status for id: {}", p.id());
+
+        p.setNextAlertStatusChange(-1);
+        switch (p.alertStatus()) {
+
+            case TESTED_POSITIVE:
+                // TODO: at present the patient doesn't move from here.. should they go to immune after some time?
+                LOGGER.trace("user {} has tested positive", p.id());
+                return alertAllContacts();
+            case AWAITING_RESULT:
+                // TODO: maybe include flag for has had virus?
+                // TODO: should a recovered person test +ve?
+                if (p.wasInfectious()) {
+                    LOGGER.trace("user {} has tested positive", p.id());
+                    updateAlertStatus(TESTED_POSITIVE, time);
+                } else {
+                    LOGGER.trace("user {} has tested negative", p.id());
+                    updateAlertStatus(NONE, time);
```

Comment:
@sam-brett  Question about this line which is confusing me a bit: It makes sense to me why we set the alert status to NONE here, however in updateAlertStatus() we then go and set the nextAlertStatusChange time to some point in the future. I believe this could then cause the alertstatus to change from NONE to ALERTED again?